handlers/filter_output.go

```diff
@@ -0,0 +1,181 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
+	"github.com/ONSdigital/dp-api-clients-go/v2/dimension"
+	"github.com/ONSdigital/dp-api-clients-go/v2/filter"
+	"github.com/ONSdigital/dp-frontend-dataset-controller/config"
+	"github.com/ONSdigital/dp-frontend-dataset-controller/helpers"
+	"github.com/ONSdigital/dp-frontend-dataset-controller/mapper"
+	"github.com/ONSdigital/dp-net/v2/handlers"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/gorilla/mux"
+)
+
+// FilterOutput will load a filtered landing page
+func FilterOutput(fc FilterClient, dimsc DimensionClient, dc DatasetClient, rend RenderClient, cfg config.Config, apiRouterVersion string) http.HandlerFunc {
+	return handlers.ControllerHandler(func(w http.ResponseWriter, req *http.Request, lang, collectionID, userAccessToken string) {
+		filterOutput(w, req, dc, fc, dimsc, rend, cfg, collectionID, lang, apiRouterVersion, userAccessToken)
+	})
+}
+
+func filterOutput(w http.ResponseWriter, req *http.Request, dc DatasetClient, fc FilterClient, dimsc DimensionClient, rend RenderClient, cfg config.Config, collectionID, lang, apiRouterVersion, userAccessToken string) {
```

Comment:
The filterOutput function is quite big. It would be good to encapsulate some of that logic in smaller methods.

Non-blocker.